src/drawtypes/iconset.cpp

```diff
@@ -20,11 +21,21 @@ namespace drawtypes {
 
     // If fuzzy matching is turned on, try that first before returning the fallback.
     if (fuzzy_match) {
+      // works by finding the *longest* matching icon to the given workspace id
+      int max_size = -1;
```

Comment:
If you make this have type `size_t`, you don't have to cast to `int` further down.

You will also have to change the condition under which you update `max_size` and `max_label` by adding `|| !max_label`. This way, the first match will update `max_size`.

Reply:
just made the necessary changes for this, thank you